clientconn.go

```diff
@@ -213,10 +216,12 @@ func WithUserAgent(s string) DialOption {
 
 // Dial creates a client connection the given target.
 func Dial(target string, opts ...DialOption) (*ClientConn, error) {
+	ctx := context.Background()
 	cc := &ClientConn{
 		target: target,
 		conns:  make(map[Address]*addrConn),
 	}
+	cc.ctx, cc.cancel = context.WithCancel(ctx)
```

Comment:
`cc.ctx, cc.cancel = context.WithCancel(context.Background())`

so that line#219 is not needed.